kratos/processes/replace_elements_and_condition_process.cpp

```diff
@@ -27,81 +27,88 @@ namespace Kratos
-    
-    #pragma omp parallel for
-    for(int i=0; i< static_cast<int>(r_root_model_part.Elements().size()); i++) {
-        auto it_elem = r_root_model_part.ElementsBegin() + i;
-        
-        auto p_element = rReferenceElement.Create(it_elem->Id(), it_elem->pGetGeometry(), it_elem->pGetProperties());
-        
-        // Deep copy elemental data and flags
-        p_element->Data() = it_elem->Data();
-        p_element->Set(Flags(*it_elem));
-        
-        (*it_elem.base()) = p_element;
+    ModelPart& r_model_part = mSettings["use_root_model_part"].GetBool() ? mrModelPart.GetRootModelPart() : mrModelPart;
+
+    const std::string element_name = mSettings["element_name"].GetString();
+    const std::string condition_name = mSettings["condition_name"].GetString();
+
+    bool replace_elements = false;
+    bool replace_conditions = false;
+
+    if (element_name != "") {
+        replace_elements = true;
+        const Element& rReferenceElement = KratosComponents<Element>::Get(mSettings["element_name"].GetString());
+        #pragma omp parallel for
+        for(int i=0; i< static_cast<int>(r_model_part.Elements().size()); i++) {
+            auto it_elem = r_model_part.ElementsBegin() + i;
+
+            auto p_element = rReferenceElement.Create(it_elem->Id(), it_elem->pGetGeometry(), it_elem->pGetProperties());
+
+            // Deep copy elemental data and flags
+            p_element->Data() = it_elem->Data();
+            p_element->Set(Flags(*it_elem));
+
+            (*it_elem.base()) = p_element;
+        }
     }
-    
-    #pragma omp parallel for
-    for(int i=0; i< static_cast<int>(r_root_model_part.Conditions().size()); i++) {
-        auto it_cond = r_root_model_part.ConditionsBegin() + i;
-        
-        auto p_condition = rReferenceCondition.Create(it_cond->Id(), it_cond->pGetGeometry(), it_cond->pGetProperties());
-        
-        // Deep copy elemental data and flags
-        p_condition->Data() = it_cond->Data();
-        p_condition->Set(Flags(*it_cond));
-        
-        (*it_cond.base()) = p_condition;
-
-    }      
-    
-    // Change the submodelparts
-    for (auto& i_sub_model_part : r_root_model_part.SubModelParts())
-        UpdateSubModelPart( i_sub_model_part, r_root_model_part );
 
-}
+    if (condition_name != "") {
+        replace_conditions = true;
+        const Condition& rReferenceCondition = KratosComponents<Condition>::Get(mSettings["condition_name"].GetString());
 
-/***********************************************************************************/
-/***********************************************************************************/
+        #pragma omp parallel for
+        for(int i=0; i< static_cast<int>(r_model_part.Conditions().size()); i++) {
+            auto it_cond = r_model_part.ConditionsBegin() + i;
 
-ModelPart& ReplaceElementsAndConditionsProcess::ObtainRootModelPart( ModelPart& rModelPart )
```

Comment:
this method already exists in the modelpart

Reply:
Yes..., sorry for that